plugins/nf-wave/src/main/io/seqera/wave/plugin/WaveClient.groovy

```diff
@@ -403,11 +434,82 @@ class WaveClient {
         return addCommands(result)
     }
 
+    // Dockerfile template adpated from the Spack package manager
+    // https://github.com/spack/spack/blob/develop/share/spack/templates/container/Dockerfile
+    // LICENSE APACHE 2.0
+    protected String spackFileToDockerFile() {
+
+        def result = """\
+# Builder image
+FROM ${config.spackOpts().builderImage} as builder
+COPY spack.yaml /tmp/spack.yaml
+
+RUN mkdir -p /opt/spack-env \\
+&&  (sed -e 's;compilers:;compilers::;' \\
+         -e 's;^ *flags: *{};    flags:\\n      cflags: ${config.spackOpts().cFlags}\\n      cxxflags: ${config.spackOpts().cxxFlags}\\n      fflags: ${config.spackOpts().fFlags};' \\
+         /root/.spack/linux/compilers.yaml) > /opt/spack-env/compilers.yaml \\
+&&  (sed '/^spack:/a  \\ \\ include: [/opt/spack-env/compilers.yaml]\\
+  concretizer:\\
+    unify: true' /tmp/spack.yaml) > /opt/spack-env/spack.yaml \\
+&&  (echo "  packages:" \\
+&&   echo "    all:" \\
+&&   echo "      target: [${config.spackOpts().target}]" \\
+&&   echo "  config:" \\
+&&   echo "    install_tree: /opt/software" \\
+&&   echo "  view: /opt/view") >> /opt/spack-env/spack.yaml
```

Comment:
What's the need for the `echo` ?

Reply:
struggling to reproduce newlines appropriately in the Dockerfile, by using a single `echo` or `cat` ..

Reply:
ok, addressed by using a single `echo` . templating (below) later on